activation/poetdb.go

```diff
@@ -52,48 +61,48 @@ func (db *PoetDb) ValidatePoetProof(proof types.PoetProof, poetId [types.PoetIdL
 	return nil
 }
 
-func (db *PoetDb) storePoetProof(proof types.PoetProof, poetId [types.PoetIdLength]byte, roundId uint64,
-	signature []byte) error {
-
-	poetProof, err := types.InterfaceToBytes(&proof)
+func (db *PoetDb) storeProof(proofMessage types.PoetProofMessage) error {
+	poetProofBytes, err := types.InterfaceToBytes(&proofMessage.PoetProof)
 	if err != nil {
 		return fmt.Errorf("failed to marshal poet proof for poetId %x round %d: %v",
-			poetId, roundId, err)
+			proofMessage.PoetId, proofMessage.RoundId, err)
 	}
 
-	ref := sha256.Sum256(poetProof)
+	ref := sha256.Sum256(poetProofBytes)
```

Comment:
it shouldn't work this way, we should have a single place that calculates references. Don't change it now but we should have a technical debt issue to handle it